components/skus/browser/rs/lib/src/storage/kv.rs

```diff
@@ -247,6 +400,44 @@ where
         Err(InternalError::StorageWriteFailed("Item credentials were not initiated".to_string()))
     }
 
+    #[instrument]
+    async fn spend_time_limited_v2_item_cred(
+        &self,
+        item_id: &str,
+        index: usize,
+    ) -> Result<(), InternalError> {
+        let mut store = self.get_store()?;
+        let mut state: KVState = store.get_state()?;
+
+        if let Some(credentials) = state.credentials.as_mut() {
+            if let Some(item_credentials) = credentials.items.get_mut(item_id) {
+                match item_credentials {
+                    Credentials::TimeLimitedV2(item_credentials) => {
+                        if let Some(creds) = item_credentials.creds.as_mut() {
+                            // iterate over each time limited cred, and find where now is between
+                            for tlv2_cred in creds {
+                                if Utc::now().naive_utc() < tlv2_cred.valid_to && Utc::now().naive_utc() > tlv2_cred.valid_from {
```

Comment:
hm I wonder i we should be worried about the time ticking over between when we select the index and spend it